backend/services/implementations/testService.ts

```diff
@@ -400,6 +407,15 @@ class TestService implements ITestService {
     );
   }
 
+  private async incrementImageCounts(
+    questions: QuestionComponent[][],
+  ): Promise<void> {
+    await this.processImages<ImageMetadata>(
+      questions,
+      this.imageUploadService.incrementImageCount.bind(this.imageUploadService),
+    );
```

Comment:
Since this runs for all images, have you done any sort of load test before/after? There's probably minimal difference considering things should theoretically run in parallel, but it would be good to verify